src/dbnode/persist/fs/write_large_tiles_test.go

```diff
@@ -0,0 +1,224 @@
+	defer os.RemoveAll(dir)
+
+	entries := []testLargeTileEntry{
+		{testEntry{"baz", nil, nil}, []float64{65536}},
+		{testEntry{"bar", nil, nil}, []float64{4.8, 5.2, 6}},
+	}
+
+	w := newTestLargeTilesWriter(t, filePathPrefix, 0, testWriterStart, 0, 5)
+	defer w.Close()
+	err := writeTestLargeTilesData(t, w, testWriterStart, entries)
+	require.Error(t, err)
+	require.Equal(t, "ids must be written in lexicographic order, no duplicates, but got baz followed by bar",
+		err.Error())
+}
+
+func TestDoubleWritesAreNotAllowed(t *testing.T) {
+	dir := createTempDir(t)
+	filePathPrefix := filepath.Join(dir, "")
+	defer os.RemoveAll(dir)
+
+	entries := []testLargeTileEntry{
+		{testEntry{"baz", nil, nil}, []float64{65536}},
+		{testEntry{"baz", nil, nil}, []float64{4.8, 5.2, 6}},
+	}
+
+	w := newTestLargeTilesWriter(t, filePathPrefix, 0, testWriterStart, 0, 5)
+	defer w.Close()
+	err := writeTestLargeTilesData(t, w, testWriterStart, entries)
+	require.Error(t, err)
+	require.Equal(t, "ids must be written in lexicographic order, no duplicates, but got baz followed by baz",
+		err.Error())
+}
+
+func TestSimpleLargeTilesReadWrite(t *testing.T) {
+	dir := createTempDir(t)
+	filePathPrefix := filepath.Join(dir, "")
+	defer os.RemoveAll(dir)
+
+	entries := []testLargeTileEntry{
+		{testEntry{"bar", nil, nil}, []float64{4.8, 5.2, 6}},
+		{testEntry{"baz", nil, nil}, []float64{65536}},
+		{testEntry{"cat", nil, nil}, []float64{100000}},
+		{testEntry{"foo", nil, nil}, []float64{1, 2, 3}},
+		{testEntry{"foo+bar=baz,qux=qaz", map[string]string{
+			"bar": "baz",
+			"qux": "qaz",
+		}, nil}, []float64{7, 8, 9}},
+	}
+
+	w := newTestLargeTilesWriter(t, filePathPrefix, 0, testWriterStart, 0, 5)
+	err := writeTestLargeTilesData(t, w, testWriterStart, entries)
+	require.NoError(t, err)
+	err = w.Close()
+	require.NoError(t, err)
+
+	expectEntries := make([]testEntry, 0, len(entries))
+	for _, e := range entries {
+		expectEntries = append(expectEntries, e.testEntry)
+	}
+
+	r := newTestReader(t, filePathPrefix)
+	readTestDataWithOrderOpt(t, r, 0, testWriterStart, expectEntries, true)
+}
+
+func TestLargeTilesInfoReadWrite(t *testing.T) {
+	dir := createTempDir(t)
+	filePathPrefix := filepath.Join(dir, "")
+	defer os.RemoveAll(dir)
+
+	entries := []testLargeTileEntry{
+		{testEntry{"bar", nil, nil}, []float64{4.8, 5.2, 6}},
+		{testEntry{"baz", nil, nil}, []float64{65536}},
+		{testEntry{"cat", nil, nil}, []float64{100000}},
+		{testEntry{"foo", nil, nil}, []float64{1, 2, 3}},
+	}
+
+	w := newTestLargeTilesWriter(t, filePathPrefix, 0, testWriterStart, 0, 12)
+	err := writeTestLargeTilesData(t, w, testWriterStart, entries)
+	require.NoError(t, err)
+	err = w.Close()
+	require.NoError(t, err)
+
+	readInfoFileResults := ReadInfoFiles(filePathPrefix, testNs1ID, 0, 16, nil, persist.FileSetFlushType)
+	require.Equal(t, 1, len(readInfoFileResults))
+	for _, result := range readInfoFileResults {
+		require.NoError(t, result.Err.Error())
```

Comment:
Since you have already established that `len(readInfoFileResults) == 1`, and are working just with `readInfoFileResults[0].Info` below, it might be more consistent to directly check `readInfoFileResults[0].Err` here as well (instrad of the loop).